packages/vite/src/node/plugins/asset.ts

```diff
@@ -137,7 +137,7 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
       // will fail to resolve in the main resolver. handle them here.
       const publicFile = checkPublicFile(id, config)
       if (publicFile) {
-        return id
+        return { id: publicFile, meta: { publicUrl: id } }
```

Comment:
So basically, the full file path is returned as the resolved ID and the original module URL is cached in `meta.publicUrl` for the `assetPlugin` to use.

Reply:
Could we call `this.addWatchFile` instead?